controllers/useraccount/console_usersettings_test.go

```diff
@@ -0,0 +1,144 @@
+	t.Run("Object found by name and deleted", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		cm := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-johnsmith",
+				Namespace: UserSettingNS,
+			},
+		}
+		cl := fake.NewClientBuilder().WithObjects(cm).Build()
+
+		// when
+		err := deleteResource(ctx, cl, "johnsmith", cm)
+
+		// then
+		require.NoError(t, err)
+		// check that the configmap doesn't exist anymore
+		err = cl.Get(ctx, client.ObjectKey{Name: "user-settings-johnsmith", Namespace: UserSettingNS}, &corev1.ConfigMap{})
+		require.True(t, errors.IsNotFound(err))
+	})
+	t.Run("Object found by label and deletes successfully", func(t *testing.T) {
+		// given
+		cm := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        "johnsmith",
+				},
+			},
+		}
+		// create a noise objects where the labels don't match
+		noiseObject := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match-noise",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+				},
+			},
+		}
+		cl := test.NewFakeClient(t, cm, noiseObject)
+
+		// when
+		err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
+
+		// then
+		require.NoError(t, err)
+		// check that the configmap doesn't exist anymore
+		AssertObjectNotFound(t, cl, UserSettingNS, "johnsmith", &corev1.ConfigMap{})
+		// check that the noise object still exists
+		retrievedNoise := &corev1.ConfigMap{}
+		AssertObject(t, cl, UserSettingNS, "user-settings-name-no-match-noise", retrievedNoise, func() {
+			assert.Equal(t, noiseObject, retrievedNoise)
+		})
+	})
+	t.Run("multiple objects found by label and deletes successfully", func(t *testing.T) {
+		// given
+		cm1 := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        "johnsmith",
+				},
+			},
+		}
+		cm2 := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match-second",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        "johnsmith",
+				},
+			},
+		}
+		cl := test.NewFakeClient(t, cm1, cm2)
+
+		// when
+		err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
```

Comment:
And here.

Reply:
I don't think this needs to be changed, since its adding the object to the fake client.